src/dynlib/dynlib_windows.c

```diff
@@ -37,6 +37,46 @@ static char *generate_file_name(const char *name, uint32_t major_ver, uint32_t m
     return versioned_file_name;
 }
 
+static DLL_DIRECTORY_COOKIE add_current_module_to_search()
+{
+    wchar_t path[MAX_PATH];
+    HMODULE hModule = NULL;
+
+    if (GetModuleHandleEx(GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS, (LPCTSTR)add_current_module_to_search, &hModule) == 0)
+    {
+        LOG_WARNING("Failed to get current module (%d).", GetLastError());
+        return NULL;
+    }
+
+    if (GetModuleFileNameW(hModule, path, sizeof(path)) == 0)
+    {
+        LOG_WARNING("Failed to get current module file name (%d).", GetLastError());
+        return NULL;
+    }
+
+    // GetModuleFileName give the full path, but AddDllDirectory requires a directory path. Remove
+    // the filename portion of this path. This should only be running in the "K4A.dll" assembly,
+    // but verify this assumption before truncating the path. If this code is used from a different
+    // file, then this needs to be modified to search for the path separators.
+    size_t length = wcslen(path);
+    wchar_t *fileName = path + (length - 7);
```

Comment:
Can you define "k4a.dll" once and use sizeof or countof to get the '7' you need here?

Reply:
I'll use Matt's suggestion as it removes the need for assuming the file name.